File: typescript/src/Testing/TestingManager.ts
/**
 * Zirve Testing Manager — Keploy
 *
 * Interfaces with Keploy UI/API endpoint for tracking test runs.
 */
export class TestingManager {
  private config: Record<string, any>;
  private url: string;

  constructor(config: Record<string, any>) {
    this.config = config;
    this.url = (config.url || 'http://keploy.zirve-infra.svc.cluster.local:6789').replace(/\/$/, '') + '/api';
  }

  private async request(method: string, path: string): Promise<any> {
    const options: RequestInit = {
      method,
      headers: { 'Content-Type': 'application/json' }
    };

    const response = await fetch(`${this.url}/${path.replace(/^\/+/, '')}`, options);
    
    if (!response.ok) {
      if (response.status === 404) return null;
      throw new Error(`Keploy API Error [${response.status}]`);
    }

    if (response.status === 204 || response.headers.get('content-length') === '0') return true;
    return await response.json();
  }

  public async listTestSets(app: string): Promise<any[]> {
    const result = await this.request('GET', `test-sets?app=${encodeURIComponent(app)}`);
    return result || [];
  }

  public async getTestRun(id: string): Promise<any | null> {
    return await this.request('GET', `test-run/${encodeURIComponent(id)}`);
  }

  public async health(): Promise<boolean> {
    try {
      // Basic ping/health based on available endpoints
      const response = await fetch(`${this.url}/healthz`); // Placeholder check
      return response.ok;
    } catch {
      return false;
    }
  }
}

File: dotnet/Zirve.Sdk/Testing/TestingManager.cs
using System;
using System.Net.Http;
using System.Text.Json;
using System.Threading.Tasks;
using Zirve.Sdk.Config;

namespace Zirve.Sdk.Testing;

public class TestingManager
{
    private readonly HttpClient _httpClient;
    private readonly string _url;

    public TestingManager(HttpClient httpClient, ConfigManager configManager)
    {
        _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
        
        var cfg = configManager.Module("testing");
        _url = (cfg.GetValueOrDefault("url", "http://keploy.zirve-infra.svc.cluster.local:6789") ?? "").TrimEnd('/') + "/api";
    }

    private async Task<JsonElement> RequestAsync(HttpMethod method, string path, object? body = null)
    {
        var request = new HttpRequestMessage(method, $"{_url}/{path.TrimStart('/')}");
        
        if (body != null)
        {
            var json = body is string s ? s : JsonSerializer.Serialize(body);
            request.Content = new StringContent(json, System.Text.Encoding.UTF8, "application/json");
        }

        var response = await _httpClient.SendAsync(request);
        
        if (response.StatusCode == System.Net.HttpStatusCode.NotFound)
            return default;

        response.EnsureSuccessStatusCode();

        if (response.StatusCode == System.Net.HttpStatusCode.NoContent || response.Content.Headers.ContentLength == 0)
            return default;

        using var doc = await JsonDocument.ParseAsync(await response.Content.ReadAsStreamAsync());
        return doc.RootElement.Clone();
    }

    public async Task<JsonElement> ListTestSetsAsync(string app)
    {
        return await RequestAsync(HttpMethod.Get, $"test-sets?app={Uri.EscapeDataString(app)}");
    }

    public async Task<JsonElement> GetTestRunAsync(string id)
    {
        return await RequestAsync(HttpMethod.Get, $"test-run/{Uri.EscapeDataString(id)}");
    }

    public async Task<bool> HealthAsync()
    {
        try
        {
            var response = await _httpClient.GetAsync($"{_url}/healthz");
            return response.IsSuccessStatusCode;
        }
        catch
        {
            return false;
        }
    }
}
